src/main/java/eu/xenit/gradle/docker/compose/DockerComposeConventionImpl.java

```diff
@@ -108,23 +124,33 @@ public void fromProject(Project project) {
                 composeTask.doFirst(CONFIGURE_COMPOSE_ACTION_NAME, new Action<Task>() { // No lambda -> see #96
                     @Override
                     public void execute(Task t) {
-                        createSetComposeEnvironmentFromPathAction().execute(dockerBuildImage);
+                        if(environmentVariable != null) {
+                            createSetComposeEnvironmentAction(environmentVariable + "_" + t.getName()).execute(dockerBuildImage);
```

Comment:
```suggestion
                            createSetComposeEnvironmentAction(environmentVariable + "_" + dockerBuildImage.getName()).execute(dockerBuildImage);
```

I made a mistake with my previous suggestion here. We want to use the name of the `dockerBuildImage` task here, not the `ComposeUp` task.

Reply:
And we need `Util.safeEnvironmentVariableName()` as well so the task name is properly converted to something that can be used for an environment variable name.